code/core/src/phone/java/com/adobe/marketing/mobile/internal/util/NetworkUtils.kt

```diff
@@ -0,0 +1,34 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.internal.util
+
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+
+/**
+ * Checks if the network is configured to reach the general Internet.
+ *
+ * @param connectivityManager the [ConnectivityManager] to use to check the network status
+ * @return `true` if the network is configured to reach the general Internet, `false` otherwise
+ */
+fun isInternetAvailable(connectivityManager: ConnectivityManager): Boolean {
```

Comment:
I'd prefer to keep it as an internal utility because it could be used in other places. Also, it's really hard to test a private static method in test code.

Reply:
Got it. In that case can we make this `internal` and add `@JvmName("isInternetAvailable")` to the method and 
`@file:JvmName("NetworkUtils")` so that you can directly use this in Java internally  (without the *Kt suffix)

Reply:
Will do.